test/api_client/api_client.js

```diff
@@ -0,0 +1,330 @@
+/*
+ * Copyright © 2017 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import APIClient, {
+	getMainnetClient,
+	getTestnetClient,
+} from 'api_client/api_client';
+
+describe('APIClient module', () => {
+	const mainnetHash =
+		'ed14889723f24ecc54871d058d98ce91ff2f973192075c0155ba2b7b70ad2511';
+	const mainnetNodes = [
+		'node01.lisk.io',
+		'node02.lisk.io',
+		'node03.lisk.io',
+		'node04.lisk.io',
+		'node05.lisk.io',
+		'node06.lisk.io',
+		'node07.lisk.io',
+		'node08.lisk.io',
+	];
+	const testnetHash =
+		'da3ed6a45429278bac2666961289ca17ad86595d33b31037615d4b8e8f158bba';
+	const testnetNodes = ['testnet.lisk.io'];
+	const defaultHeaders = {
+		'Content-Type': 'application/json',
+		nethash: mainnetHash,
+		os: 'lisk-js-api',
+		version: '1.0.0',
+		minVersion: '>=1.0.0',
+	};
+
+	const customHeaders = {
+		'Content-Type': 'application/json',
+		nethash: testnetHash,
+		os: 'lisk-js-api',
+		version: '0.5.0',
+		minVersion: '>=0.1.0',
+	};
+
+	const localNode = 'http://localhost:7000';
+	const externalNode = 'https://googIe.com:8080';
+	const sslNode = 'https://external.lisk.io:443';
+	const externalTestnetNode = 'http://testnet.lisk.io';
+	const defaultNodes = [localNode, externalNode, sslNode];
+	const defaultSelectedNode = 'selected_node';
+
+	let apiClient;
+
+	beforeEach(() => {
+		apiClient = new APIClient(defaultNodes, mainnetHash);
+	});
+
+	describe('#constructor', () => {
+		it('should throw an error if no arguments are passed to constructor', () => {
+			return (() => new APIClient()).should.throw(
+				Error,
+				'Require nodes to be initialized.',
+			);
+		});
+
+		it('should throw an error if first argument passed to constructor is not array', () => {
+			return (() => new APIClient('non-array')).should.throw(
+				Error,
+				'Require nodes to be initialized.',
+			);
+		});
+
+		it('should throw an error if first argument passed to constructor is empty array', () => {
+			return (() => new APIClient([])).should.throw(
+				Error,
+				'Require nodes to be initialized.',
+			);
+		});
+
+		it('should throw an error if no second argument is passed to constructor', () => {
+			return (() => new APIClient(defaultNodes)).should.throw(
+				Error,
+				'Require nethash to be initialized.',
+			);
+		});
+
+		it('should throw an error if second argument is passed to constructor is not string', () => {
```

Comment:
`should throw an error if second argument passed to constructor is not string`. Also applies to other tests.